### PR TITLE
[vm] Fix tracking for VM_MAX_VALUE_DEPTH_REACHED

### PR DESCRIPTION
**Pull request by tnowacki**
- Made type layout generation eager in data store for events and resources
- The eagerness moves the error from change set generation to the instruction that causes the change, thus giving location information

## Motivation

- Making VM_MAX_VALUE_DEPTH_REACHED fit into the execution failure bucket 

## Test Plan

- New test for events

### CODE REVIEW

**Comment by tnowacki** [comment]
This is "technically" a breaking change, as the offsets for execution failures for VM_MAX_VALUE_DEPTH_REACHED will be different now. But this error will never be reached in any of the code we currently have. 

How do we want to approach this?

**Comment by bob-wilson** [comment]
> How do we want to approach this?

I'm generally inclined to be conservative about what we consider to be breaking changes, but in this case, it really seems like overkill. Maybe someone else will think of a case where it could matter, but I can't come up with anything. I'd be OK with not treating this as breaking if no one objects.

File: language/move-lang/functional-tests/tests/natives/lcs_large_value.move [hunk 1]
```diff
@@ -84,4 +84,4 @@ script {
         M::encode_257();
     }
 }
-// check: ABORTED { code: 453
+// check: ABORTED { code: 453, location: 00000000000000000000000000000001::LCS }
```
Comment (vgao1996) [inline language/move-lang/functional-tests/tests/natives/lcs_large_value.move:87]: Just a side note:
```
// check: ABORTED { code: 453, location: 00000000000000000000000000000001::LCS }
// gets desugared into 
// check: ABORTED
// check: {
// check: 453,
// check: location:
// check: 00000000000000000000000000000001::LCS
// check: }
```

To be clear, this was a terrible design choice I made at the time and the whole checker is likely going away soon. For now I'd suggest quoting the whole thing.
```// check: "ABORTED { code: 453, location: 00000000000000000000000000000001::LCS }"```

**Pull request closed without merging**